src/app/icd/client/CheckInHandler.h

```diff
@@ -62,7 +72,20 @@ class CheckInMessageHandler : public Messaging::ExchangeDelegate, public Messagi
      * @param[in] clientInfo clientInfo object
      * @param[in] keyData New key data to use to re-register the client with the server
      */
-    CHIP_ERROR RegisterClientWithNewKey(ICDClientInfo & clientInfo, const ByteSpan keyData);
+    CHIP_ERROR RegisterClientWithNewKey(ICDClientInfo & clientInfo, ByteSpan newKey, Messaging::ExchangeManager & exchangeMgr,
+                                        const SessionHandle & sessionHandle);
+
+    /**
+     * @brief Sets up a CASE session to the peer, if we can locate a
```

Comment:
For readability, we may particularly describe the purpose to create the case session in CheckInHandler is to used to send RegisterClient command when refresh new key is necessary..?

Reply:
Okay, will add more details.